File: Data Structures & Algorithms/longest-common-prefix/submission-3.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort(key = len)
        res = strs[0]
        
        for s in strs[1:]:
            for i in range(len(res)):
                if s[i] != res[i]:
                    res = res[:i]
                    break

        return res

File: Data Structures & Algorithms/anagram-groups/submission-3.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> anagrams;

        for (string s: strs) {
            vector<int> count(26, 0);
            for (char c: s) {
                count[c - 'a']++;
            }
            string hashKey = to_string(count[0]);
            for (int i = 1; i < 26; ++i) {
                hashKey += ',' + to_string(count[i]);
            }
            anagrams[hashKey].push_back(s);

        }

        vector<vector<string>> res;
        for (auto& pair: anagrams) {
            res.push_back(pair.second);
        }
        return res;
    }
};
